docs/2.guide/2.directory-structure/1.plugins.md

```diff
@@ -166,6 +167,15 @@ export default defineNuxtPlugin(() => {
   }
 })
 ```
+```ts [plugins/hello-object-syntax.ts]
+export default defineNuxtPlugin({
+  name: "hello",
+  setup(nuxtApp) {
+    nuxtApp.provide('hello', (msg: string) => `Hello ${msg}!`)
```

Comment:
I wouldn't recommend this, but instead return an object, which allows type inference.

Is there a reason that isn't working for you?

Reply:
You are right! My issue was that in my reproduction/test I had `async setup` instead of `setup`. With async, typescript said that the setup should return `void`, which confused me. Should we mention that too in the docs you think?

Reply:
I fixed that with the latest commit.

Reply:
it should allow a return even if setup is async...

Reply:
I'll create a reproduction and if that's the case I'll open an issue :)

Reply:
Nope! not a bug! It my my mistake in my project. Somewhere in the plugin, I was doing `if(..) return`, which caused the typescript to go mad at me!